tlvccmd/disk_utils/mkfs.c

```diff
@@ -309,10 +314,22 @@ int main(int argc, char ** argv)
 		die("unable to open device");
 	if (fstat(DEV,&statbuf)<0)
 		die("unable to stat %s");
-	//else if (statbuf.st_rdev == 0x0300 || statbuf.st_rdev == 0x0340)
-		//die("Will not try to make filesystem on '%s'");
-
+	if (!(buf = malloc(512))) 
+		die("Cannot malloc buffer memory"); 
+	end = lseek(DEV, 0, SEEK_END);
```

Comment:
Instead of `lseek`, why not just use `statbuf.st_size` which has the same value from above?

Also, why is an additional `lseek` and actual `read` required below, rather than just believing the block device inode size reported? It would seem to me either that the inode size (set by the driver open routine) should be enforced by the driver `do_request` function anyways? 

I know that presently there may be the case that not all block drivers have the max size check (comparing req->rq_sector with max sectors) present in the block driver.... but if that's the case then the `read` below would succeed anyways.

Reply:
Thanks @ghaerr,
Statbuf.st_size - of course, I forgot. 

The rest is trial and error. The extra lseek and read is insurance, the max (SEEK_END) turns out to be unreliable (for now) (floppies).

Of course this should (and will) be fixed driver level - for now this helps avoiding mistakes. And mkfs becomes the litmus test when the low level supposedly works.